SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1579,11 +1585,51 @@ bool ExtractComponentSetsFromSchemaJson(const FString& SchemaJsonPath, TMap<uint
 	TMap<FString, uint32> ComponentMap;
 	TMap<uint32, TSet<FString>> ComponentRefSetMap;
 
+	TMap<FString, uint32> DataDefinitionNameToTypeId;
+	TMap<uint32, FString> ComponentIdToDataDefinitionName;
+
 	for (const auto& FileValue : *SchemaFiles)
 	{
 		const TSharedPtr<FJsonObject>* FileObject;
 		SAFE_TRYGET(FileValue, Object, FileObject);
 
+		const TArray<TSharedPtr<FJsonValue>>* TypesDecl;
+		SAFE_TRYGETFIELD((*FileObject), Array, "types", TypesDecl);
+
+		for (const auto& TypeValue : *TypesDecl)
+		{
+			const TSharedPtr<FJsonObject>* TypeObject;
+			SAFE_TRYGET(TypeValue, Object, TypeObject);
+
+			FString ComponentName;
+			SAFE_TRYGETFIELD((*TypeObject), String, "qualifiedName", ComponentName);
+
+			uint32 TypeId = OutTypeIdToFieldIdsMap.Num();
```

Comment:
If typeid is just incrementing - why not just use an array?

Reply:
True, I guess the only thing I am worried about is the readability of the code.
If you think it's ok, then I can change it.

Reply:
I think it might simplify it somewhat, you could then remove TypeId and replace it with something you call FieldIdIdx, which is more explicit.

Reply:
Did this now, can you take a look?